packages/pubsub/src/Providers/AWSAppSyncRealTimeProvider.ts

```diff
@@ -142,6 +142,11 @@ const START_ACK_TIMEOUT = 15000;
  */
 const DEFAULT_KEEP_ALIVE_TIMEOUT = 5 * 60 * 1000;
 
+const standardDomainPattern =
+	'^https://\\w{26}.appsync-api.\\w{2}(?:(?:-\\w{2,})+)-\\d.amazonaws.com/graphql$';
```

Comment:
This pattern is not precise as defined. I can insert any character in place of the periods (`.`) and still get it to match. We should escape it -> `\.`. This is [called out](https://lgtm.com/projects/g/aws-amplify/amplify-js/rev/pr-8560610adab5295d45c45daa006d5278cf7389ec) by LGTM as well.

For example, the following value will return a successful match, even though it's clearly incorrect (`+` instead of `.`):
`https://xxxxxxxxxxxxxxxxxxxxxxxxxx+appsync-api+us-east-2+amazonawsscom/graphql`

Overall, I think we should use a regular expression literal here (see suggested change), instead of a string pattern. Alternatively, we can construct a RegExp object by passing it the string pattern.

```suggestion
	/^https:\/\/\w{26}\.appsync-api\.\w{2}(?:(?:-\w{2,})+)-\d\.amazonaws.com\/graphql/
```

Reply:
Thanks Ivan, this is a great catch :)